packages/core/bootstrap/src/lib/cache/index.ts

```diff
@@ -175,7 +177,19 @@ export const withCache: Middleware = async (execute, options = defaultOptions())
         logger.debug(`Cache: GET ${key}`, entry)
         const reqMaxAge = _getRequestMaxAge(data)
         if (reqMaxAge && reqMaxAge !== entry.maxAge) await _cacheOnSuccess(entry)
-        return { jobRunID: data.id, ...entry }
+        const ttl = await cache.pttl(key)
+        const staleness = entry.maxAge - ttl
+        const debug = {
+          cacheHit: true,
+          staleness,
+          performance: endMetrics(true, staleness),
+          providerCost: 0,
+        }
+        return {
+          jobRunID: data.id,
+          ...entry,
+          debug,
```

Comment:
We should make sure we don't override the debug info other parts of the system added.